platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/internal/NoOpTaskScheduler.java

```diff
@@ -0,0 +1,130 @@
+ *              just to appease the compiler, as this scheduler never produces output.
+ */
+public class NoOpTaskScheduler<OUT> extends TaskScheduler<OUT> {
+
+    private final StandardWiringModel model;
+
+    /**
+     * Constructor.
+     *
+     * @param model               the wiring model containing this task scheduler
+     * @param name                the name of the task scheduler
+     * @param type                the type of task scheduler
+     * @param flushEnabled        if true, then {@link #flush()} will be enabled, otherwise it will throw.
+     * @param squelchingEnabled   if true, then squelching will be enabled, otherwise trying to squelch will throw.
+     */
+    public NoOpTaskScheduler(
+            @NonNull final StandardWiringModel model,
+            @NonNull final String name,
+            @NonNull final TaskSchedulerType type,
+            final boolean flushEnabled,
+            final boolean squelchingEnabled) {
+        super(model, name, type, flushEnabled, squelchingEnabled, false);
+
+        this.model = Objects.requireNonNull(model);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getUnprocessedTaskCount() {
+        return 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void flush() {
+        throwIfFlushDisabled();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void put(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
+        throw new UnsupportedOperationException(
+                "Data should have been discarded before being sent to this no-op scheduler");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean offer(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
+        throw new UnsupportedOperationException(
+                "Data should have been discarded before being sent to this no-op scheduler");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void inject(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
+        throw new UnsupportedOperationException(
+                "Data should have been discarded before being sent to this no-op scheduler");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    protected StandardOutputWire<OUT> buildPrimaryOutputWire(
+            @NonNull final StandardWiringModel model, @NonNull final String name) {
+        return new NoOpOutputWire<>(model, getName());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public <T> StandardOutputWire<T> buildSecondaryOutputWire() {
+        return new NoOpOutputWire<>(model, getName());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public <I> BindableInputWire<I, OUT> buildInputWire(@NonNull final String name) {
+        return new NoOpInputWire<>(model, this, name);
+    }
+}
```

Comment:
Do you see being used in production code or is the sole purpose for tests? If it's for Turtle only, it might be useful to isolate turtle related 'tools' in some separate package where they're appropriately marked.

Reply:
There is a production use case. When we have a component that we may want to disable via configuration, a no op scheduler is a super useful utility. In this PR, if you look inside `PlatformWiring.java`, I use a no-op scheduler when we want to disable the `PlatformPublisher` component.